### PR TITLE
feat: Added chat integration v2

### PR DESCRIPTION
**Pull request by MateuszOssGit**
Added chat integration v2

### CODE REVIEW

File: libs/ibm/langchain_ibm/chat_models.py [hunk 1]
```diff
@@ -47,22 +50,25 @@
     ToolCall,
     ToolMessage,
     ToolMessageChunk,
-    convert_to_messages,
```
Comment (Mateusz-Switala) [inline libs/ibm/langchain_ibm/chat_models.py:50]: It seems that LangSmithParams should be rather imported from `langchain_core.language_models.base`

File: libs/ibm/langchain_ibm/chat_models.py [hunk 2]
```diff
@@ -746,47 +629,41 @@ def _create_message_dicts(
         message_dicts = [_convert_message_to_dict(m) for m in messages]
         return message_dicts, params
 
-    def _create_chat_result(self, response: Union[dict]) -> ChatResult:
+    def _create_chat_result(
+        self, response: Union[dict], generation_info: Optional[Dict] = None
+    ) -> ChatResult:
```
Comment (Mateusz-Switala) [inline libs/ibm/langchain_ibm/chat_models.py:634]: `Union` requires two or more type arguments

File: libs/ibm/langchain_ibm/chat_models.py (hunk 1 above)
Comment (MateuszOssGit) [reply libs/ibm/langchain_ibm/chat_models.py:52]: removed

File: libs/ibm/langchain_ibm/chat_models.py (hunk 2 above)
Comment (MateuszOssGit) [reply libs/ibm/langchain_ibm/chat_models.py:634]: removed

**Pull request merged**